pkg/arm/v10/resources.go

```diff
@@ -628,5 +680,24 @@ func vmss(cs *api.OpenShiftManagedCluster, app *api.AgentPoolProfile, backupBlob
 		}
 	}
 
-	return vmss, nil
+	resource := &arm.Resource{
+		Resource:   vmss,
+		APIVersion: apiVersions["compute"],
+	}
+
+	if app.Role == api.AgentPoolProfileRoleMaster {
```

Comment:
I'm doubting whether it makes sense to remove FixupDepends (at the very least, as part of this PR).  Removing it gives more opportunity for manual error.

Reply:
I hate `FixupDepends` and `FixupApiVersions`, but I agree: it makes maintanance of our ARM templates harder.

Reply:
I reverted `FixupDepends` and `FixupApiVersions` changes